### ISSUE TITLE
Decoupling Common and Configuration

### ISSUE THREAD

**Issue by ktf** [AliceO2Group/AliceO2#836]
I am trying to have @knopers8 stuff integrated and one of the problems is that `Configuration` has an implicit dependency on `Common`. While we can workaround this by adding an extra dependency it seems to me that we should really decouple the two by moving the utilities which current depend on `Common` outside `Configuration` and leave `Configuration` as a pure API with no dependency on `libCommon.dylib`.

So I would suggest that we move:

https://github.com/AliceO2Group/Configuration/tree/master/src/CommandLineUtilities

to `AliceO2` and we drop the dependency of `Configuration` on `Common` (AFAICT, that's the only place where such a dependency happens, due to `AliceO2::Program` which by the way should not be a "common" dependency, just like `FairMQDevice` is not).

Basically my point is always the usual one, our dependency chain should look like:

<img width="1577" alt="dependencies" src="https://user-images.githubusercontent.com/10544/35908849-1d02f756-0bf2-11e8-8fec-3ced31e97da8.png">

not, like now:

<img width="1478" alt="deepdeps" src="https://user-images.githubusercontent.com/10544/35909212-249ba804-0bf3-11e8-8a28-a520f71ebcab.png">

I.e. the dependency tree should be wide and loosely coupled, not deep and strongly coupled because the latter makes things difficult to maintain longer term.

**Comment by ktf** [comment]
See also AliceO2Group/Configuration#3

**Comment by awegrzyn** [comment]
This can be closed, `Configuration` does not depend on `Common` anymore.
